node.go

```diff
@@ -216,6 +241,10 @@ func (n *ProtoNode) Copy() format.Node {
 	if len(n.links) > 0 {
 		nnode.links = make([]*format.Link, len(n.links))
 		copy(nnode.links, n.links)
```

Comment:
Same `append` note (I know you didn't modified that).
Imo just call `slices.Clone` we can now move to go1.19.